winpr/libwinpr/path/shell.c

```diff
@@ -406,19 +407,39 @@ char* GetCombinedPath(const char* basePath, const char* subPath)
 		CopyMemory(path, basePath, basePathLength);
 	path[basePathLength] = '\0';
 
-	PathCchConvertStyleA(path, basePathLength, PATH_STYLE_NATIVE);
+	if (PathCchConvertStyleA(path, basePathLength, PATH_STYLE_NATIVE) != S_OK)
+	{
```

Comment:
Even if the check for S_OK if it is correct most of the time please use the FAILED() or SUCCEEDED() macro for testing HRESULT values.